plugins/mattermost/server/api/mattermost.ts

```diff
@@ -0,0 +1,64 @@
+import Router from "koa-router";
+import { IntegrationService, IntegrationType, UserRole } from "@shared/types";
+import auth from "@server/middlewares/authentication";
+import { transaction } from "@server/middlewares/transaction";
+import validate from "@server/middlewares/validate";
+import { Integration, IntegrationAuthentication } from "@server/models";
+import { presentIntegration, presentPolicies } from "@server/presenters";
+import { APIContext } from "@server/types";
+import * as T from "./schema";
+
+const router = new Router();
+
+router.post(
+  "mattermost.connect",
```

Comment:
Added a new endpoint, instead of using `integrations.create`, since we need to store the api-key securely in `IntegrationAuthentication` model.

Reply:
I think this is a pretty common case actually, how about we update `integrations.create` to allow passing optional authentication token instead of an entire new endpoint. Less maintenance, thoughts?

Reply:
True - thought this might be a one-off case.